### PR TITLE
Biological Data Science abstract

### PR DESCRIPTION
**Pull request by morgantaschuk**
@gvwilson : if you could look at this before 11:59 pm today, I'd be ever so grateful.

I'm on the fence about including citations in abstracts. If you have opinions, I'm open to them.

### CODE REVIEW

File: abstract.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Making research software more robust
+Morgan Taschuk and Greg Wilson
+
+
+At the cutting edge of research, bioinformatics software is often written by a
```
Comment (gvwilson) [inline abstract.md]: I would remove the opening clause - this is true everywhere, not just at the bleeding edge.

File: abstract.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+single person who is both developer and scientist, producing insightful, novel
```
Comment (gvwilson) [inline abstract.md]: "intent on producing" or "with the primary aim of producing"

File: abstract.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+results for their own projects. Upon presenting their work, other
+people will want to use their methods. But much of academic software is written
```
Comment (gvwilson) [inline abstract.md]: "...for their own projects.  Other people may want to use their methods, but much of academic software..." (basically chopping out a sentence and stitching two others together)

File: abstract.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+in a prototypic, exploratory and rigid style that is incompatible with reuse. Often, the end user will
```
Comment (gvwilson) [inline abstract.md]: "in an ad hoc, exploratory style" (I don't think most people will understand "prototypic", and the sense of "rigid" won't be clear until you've introduced "robust" as contrast)

File: abstract.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+need to spend a considerable amount of time either reading code or in
+communication with the original developers. More times than we like to admit,
```
Comment (gvwilson) [inline abstract.md]: "...with the original developer(s) in order to get the software running."

File: abstract.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+we will simply give up on the software, either writing our own or finding
```
Comment (gvwilson) [inline abstract.md]: "we will simply give up and either write our own or find something else that accomplishes..."

File: abstract.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+another that accomplishes the same purpose. This practice is detrimental to
+creating reproducible results, to the pace of scientific research, and to
+creating bioinformatics communities that build on each other's work.
```
Comment (gvwilson) [inline abstract.md]: replace "creating...other's work" with "sharing practical knowledge within our community".

File: abstract.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+This familiar problem has spawned a range of solutions, from virtual machines
+and containerization[1], to more training for bioinformaticians
```
Comment (gvwilson) [inline abstract.md]: space before `[1]`

File: abstract.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+[2], to reform of scientific reward [3]. All of
```
Comment (gvwilson) [inline abstract.md:18]: "...[3], all of which require time and money." (I'd drop "...and none directly...")

File: abstract.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+these solutions require time and money, and none directly contribute to
+scientific advancement. What can the front-line bioinformatician do right now
```
Comment (gvwilson) [inline abstract.md]: I would cut the rhetorical question and use, "As an alternative, we present a set of guidelines..." (joining up the two paras)

**Comment by gvwilson** [comment]
Comments added - I'd include the citation links (they're short).  Thanks again for putting this together.

**Comment by morgantaschuk** [comment]
Fantastic, thanks.

**Pull request merged**